pycbc/workflow/versioning.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2023 Gareth Cabourn Davies
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+Module to generate/manage the executable used for version information
+in workflows
+"""
+
+from urllib.request import pathname2url
+from urllib.parse import urljoin
+from pycbc.workflow.core import File, FileList, Executable
+
+
+class VersioningExecutable(Executable):
+    """
+    Executable for getting version information
+    """
+    current_retention_level = Executable.FINAL_RESULT
+
+
+def make_versioning_page(workflow, config_parser, out_dir, tags=None):
+    """
+    Make executable for versioning information
+    """
+    vers_exe = VersioningExecutable(
+        workflow.cp,
+        'page_versioning',
+        out_dir=out_dir,
+        ifos=workflow.ifos,
+        tags=tags,
+    )
+    node = vers_exe.create_node()
+    exe_names = []
+    exe_paths = []
+    for name, path in config_parser.items('executables'):
+        file_url = urljoin('file:', pathname2url(path))
+        exe_to_test = File(workflow.ifos, '',
+                           workflow.analysis_time, file_url=file_url)
```

Comment:
This is not the right approach. It will declare the executables as *separate* files to the executables *already* declared in the workflow.

We might want to just ignore pegasus data handling here, unless you want to explore how pegasus will handle Executables as files in jobs. This then means that we do not try to run this executable on something like OSG, and just send it a list of URLs to files.

An additional problem at this stage would be the inspiral file in a singularity container, which will break this block of code.